### ISSUE TITLE
get_progress endpoint allows retrieval of arbitrary tasks

### ISSUE THREAD

**Issue by hansegucker** [czue/celery-progress#83]
Any user, independent whether they are logged in or permitted, can get data about any task using the get_progress endpoint.

**Comment by OmarWKH** [comment]
To do this I wrapped the `get_progress` view in my own view that requires authorization. My view would take a task id, does authorization, then forward the id to `celery_progress.views.get_progress` and return its result.

The authorization step depends on how your project is set up. Just protect your view like any other.

And in `urls.py`, I hooked up my own view instead of `celery_progress.urls`.

**Comment by czue** [comment]
Agreed with the above. Though i do wonder if there's a way to improve the situation out-of-the-box via a few settings or something. certainly at least adding a setting to require auth would be trivial. I think limiting it to the specific user would be possible, but a bit more work.

At a minimum I think this behavior should be more clearly documented so that it doesn't trip people up.

**Comment by hansegucker** [comment]
First of all, thanks for your fast responses 👍🏻. 

@OmarWKH Do you limit the task fetching to specific users or do you allow it just for all logged-in users? If yes, it would be nice to know how you save the information which task belongs to which user.

@czue I think some kind of information like the one provided by @OmarWKH would be already fine.

**Comment by OmarWKH** [comment]
@hansegucker Specific users.

I had a Task model that stored users and tasks in the database. I made sure to delete the row and forget the task when I'm done with it.

I wrote the code a while ago so I don't want to give details that I could be misinterpreting. Also my solution was a bit different to fit my use case. So here are some notes that might not be applicable:

- I used `transaction.on_commit` in the view to start the task.

- I pre-generated the task id so I can store it before the task starts.

- I used `on_success` and `on_failure` in the task to call a function that updates a model I had. That functions schedules a task that 1) deletes the Task row and 2) forgets the original task after a while.

This never went to production so I don't know if this is the best way to do it.

#50 has a relevant discussion.

**Comment by czue** [comment]
Very welcome!
